mifosng-provider/src/main/java/org/mifosplatform/portfolio/group/service/GroupingTypesWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -185,6 +186,16 @@ private CommandProcessingResult createGroupingType(final JsonCommand command, fi
 
             // pre-save to generate id for use in group hierarchy
             this.groupRepository.save(newGroup);
+            
+			/* Generate Hierarchy for groups associated to center */
+			if (newGroup.isCenter() && newGroup.hasActiveGroups()) {
+				for (final Group group : groupMembers) {
+					Long groupId = group.getId();
+					groupForUpdate = this.groupRepository
+							.findOneWithNotFoundDetection(groupId);
+					groupForUpdate.generateHierarchyToAssociateGroups(newGroup);
+				}
+			}          
 
             newGroup.generateHierarchy();
```

Comment:
@chandrikamohith : Given that there already exists a method to generate hierarchy for a group (the above line) , the logic for updating hierarchy of child groups could have been encapsulated within the same